### PR TITLE
v2 api: stats improvements

### PR DESCRIPTION
**Pull request by vrothberg**
This PR includes a series of commits to:
* Clean up.
* Comment the code.
* Fix errors and perform rigorous error checks instead of ignoring some.
* Enable the libpod endpoint (uses the generic one).
* Expect that the container is running and error out otherwise.

### CODE REVIEW

**Comment by vrothberg** [comment]
@jwhonce @baude PTAL

**Comment by baude** [comment]
this iss good work ... lgtm

File: pkg/api/handlers/generic/containers_stats.go [hunk 1]
```diff
@@ -43,28 +43,26 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// If the container isn't returning, then let's not bother and return
```
Comment (mheon) [inline pkg/api/handlers/generic/containers_stats.go]: s/returning/running

**Comment by mheon** [comment]
One nit, otherwise LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**